src/lightweight_helper.js

```diff
@@ -0,0 +1,31 @@
+
+/**
+ * @param {String} path
+ * @param {String} integrity
+ */
+export const loadScript = (path, integrity) => {
+  // eslint-disable-next-line
+    if(self.importScripts) {
+    return importScripts(path);
+  }
+  return new Promise((resolve, reject) => {
+    const script = document.createElement('script');
+    script.src = path;
+    if (integrity) {
+      script.integrity = integrity;
+    }
+    script.onload = e => resolve(e);
+    script.onerror = e => reject(e);
```

Comment:
```suggestion
    script.onerror = () => reject(new Error('Failed to load ' + path));
```

Reply:
I have researched a bit error that are thrown here, they are mostly network related.  This promise would be resolved even if path is wrong, and actually elliptic has failed to load.

Reply:
Example here https://codesandbox.io/s/intelligent-darwin-w9cxx

Reply:
Ok. But `e` is not an error object, it is an event object. You should throw an error instead of an event. Even for network errors, "failed to load" is pretty descriptive.

Reply:
It is actually error event is passed here, I would better catch it in loadEllipticPromise.

Reply:
It is an error event, but not an error object. You could do
```js
    script.onerror = e => reject(new Error(e.message));
```
if you prefer, but you should always throw an error object.